opm/autodiff/SimulatorBase_impl.hpp

```diff
@@ -682,17 +682,20 @@ namespace Opm
 
     template <class Implementation>
     V
-    SimulatorBase<Implementation>::FIPTotals(const std::vector<V>& fip, const std::vector<double>& press)
+    SimulatorBase<Implementation>::FIPTotals(const std::vector<V>& fip, const ReservoirState& state)
     {
         V totals(V::Zero(7));
         for (int i = 0; i < 5; ++i) {
             for (size_t reg = 0; reg < fip.size(); ++reg) {
                 totals[i] += fip[reg][i];
             }
         }
-        const V p = Eigen::Map<const V>(& press[0], press.size());
-        totals[5] = geo_.poreVolume().sum();
-        totals[6] = unit::convert::to((p * geo_.poreVolume()).sum() / totals[5], unit::barsa);
+        const int nc = Opm::AutoDiffGrid::numCells(grid_);
+        const int np = state.numPhases();            
+        const DataBlock s = Eigen::Map<const DataBlock>(& state.saturation()[0], nc, np);
+        const V p = Eigen::Map<const V>(& state.pressure()[0], nc);
+        totals[5] = (geo_.poreVolume() * (s.col(Oil) + s.col(Gas))).sum();
```

Comment:
> because it is wrong for two-phase...

I think for two-phase, this is wrong, will push a fix.

>I meant a variable in a programming sense, local to this function.

Now I see.